### PR TITLE
KUBESAW-43: Replace the special ToolchainClusterCondition with the standard toolchain Condition

### PR DESCRIPTION
**Pull request by fbm3307**
This is to replace the special ToolchainClusterCondition with the standard toolchain Condition

This is Related to Change in

- API - https://github.com/codeready-toolchain/api/pull/416
- Host-Operator - https://github.com/codeready-toolchain/host-operator/pull/1017
- Member-Operator - https://github.com/codeready-toolchain/member-operator/pull/560
- Toolchain-Common - https://github.com/codeready-toolchain/toolchain-common/pull/391
- KSCTL - https://github.com/kubesaw/ksctl/pull/38

Only change is updating the dependency of API and toolchain-common

- Registration-Service - https://github.com/codeready-toolchain/registration-service/pull/424
- Sandbox-SRE - https://github.com/codeready-toolchain/sandbox-sre/pull/1706

### CODE REVIEW

File: setup/users/create_users.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func getMemberClusterName(cl client.Client, hostOperatorNamespace, memberOperato
 	return memberCluster.Name, err
 }
 
-func containsClusterCondition(conditions []toolchainv1alpha1.ToolchainClusterCondition, contains *toolchainv1alpha1.ToolchainClusterCondition) bool {
+func containsClusterCondition(conditions []toolchainv1alpha1.Condition, contains *toolchainv1alpha1.Condition) bool {
```
Comment (MatousJobanek) [inline setup/users/create_users.go:72]: We don't need this special function anymore, do we? We can use the other generic function used for the standard Toolchain condition

Comment (fbm3307) [reply setup/users/create_users.go:72]: I didn't understand it completely. Does this mean to remove this function? i dont see a generic function though which can be used in its place..? and hence i kept this

Comment (MatousJobanek) [reply setup/users/create_users.go:72]: yes, you can remove this function and change the place where it is being called to:
```go
if condition.IsTrue(cluster.Status.Conditions, toolchainv1alpha1.ToolchainClusterReady) {
	memberCluster = cluster
	return true, nil
}
```
`condition.IsTrue` is the generic function I was referring to

Comment (fbm3307) [reply setup/users/create_users.go:72]: ohk got it, i somehow thought there is generic function for standard toolchain condition, which i couldn't find and hence the confusion , thanks for clarifying

File: testsupport/wait/awaitility.go [hunk 2]
```diff
@@ -161,16 +156,16 @@ func (a *Awaitility) WaitForService(t *testing.T, name string) (corev1.Service,
...
-func (a *Awaitility) WaitForToolchainClusterWithCondition(t *testing.T, namespace string, condition *toolchainv1alpha1.ToolchainClusterCondition) (toolchainv1alpha1.ToolchainCluster, error) {
+func (a *Awaitility) WaitForToolchainClusterWithCondition(t *testing.T, namespace string, cdtype toolchainv1alpha1.ConditionType) (toolchainv1alpha1.ToolchainCluster, error) {
 	t.Logf("waiting for ToolchainCluster in namespace '%s'", namespace)
 	timeout := a.Timeout
-	if condition != nil {
+	if cdtype != "" {
 		timeout = ToolchainClusterConditionTimeout
 	}
 	var c toolchainv1alpha1.ToolchainCluster
 	err := wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
```
Comment (MatousJobanek) [inline testsupport/wait/awaitility.go:166]: The function is not actually called without the ConditionType parameter specified, so this if statement doesn't make much sense anymore. 
```suggestion
	var c toolchainv1alpha1.ToolchainCluster
	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
```

File: testsupport/wait/awaitility.go [hunk 3]
```diff
@@ -180,10 +175,10 @@ func (a *Awaitility) WaitForToolchainClusterWithCondition(t *testing.T, namespac
...
 // and with the given ClusterCondition (if it the condition is nil, then it skips this check)
-func (a *Awaitility) WaitForNamedToolchainClusterWithCondition(t *testing.T, name string, condition *toolchainv1alpha1.ToolchainClusterCondition) (toolchainv1alpha1.ToolchainCluster, error) {
-	t.Logf("waiting for ToolchainCluster '%s' in namespace '%s' to have condition '%v'", name, a.Namespace, condition)
+func (a *Awaitility) WaitForNamedToolchainClusterWithCondition(t *testing.T, name string, cdtype toolchainv1alpha1.ConditionType) (toolchainv1alpha1.ToolchainCluster, error) {
+	t.Logf("waiting for ToolchainCluster '%s' in namespace '%s' to have condition '%v'", name, a.Namespace, cdtype)
 	timeout := a.Timeout
-	if condition != nil {
+	if cdtype != "" {
 		timeout = ToolchainClusterConditionTimeout
 	}
```
Comment (MatousJobanek) [inline testsupport/wait/awaitility.go:183]: the same as above - this can be removed

File: testsupport/wait/awaitility.go [hunk 4]
```diff
@@ -180,10 +175,10 @@ func (a *Awaitility) WaitForToolchainClusterWithCondition(t *testing.T, namespac
 
 // WaitForNamedToolchainClusterWithCondition waits until there is a ToolchainCluster with the given name
 // and with the given ClusterCondition (if it the condition is nil, then it skips this check)
-func (a *Awaitility) WaitForNamedToolchainClusterWithCondition(t *testing.T, name string, condition *toolchainv1alpha1.ToolchainClusterCondition) (toolchainv1alpha1.ToolchainCluster, error) {
-	t.Logf("waiting for ToolchainCluster '%s' in namespace '%s' to have condition '%v'", name, a.Namespace, condition)
+func (a *Awaitility) WaitForNamedToolchainClusterWithCondition(t *testing.T, name string, cdtype toolchainv1alpha1.ConditionType) (toolchainv1alpha1.ToolchainCluster, error) {
```
Comment (MatousJobanek) [inline testsupport/wait/awaitility.go:178]: not related to your changes in your PR, but this function doesn't seem to be used anywhere. If it's true, then we can easily remove it

**Review by MatousJobanek — APPROVED** [review]
overall looks good, just a suggestion for simplification

File: testsupport/wait/awaitility.go (hunk 2 above)
Comment (fbm3307) [reply testsupport/wait/awaitility.go:166]: yea removed it now
Had thought about it, but wasnt sure about why we were setting a different timeout and hence had left it,

File: testsupport/wait/awaitility.go (hunk 4 above)
Comment (fbm3307) [reply testsupport/wait/awaitility.go:178]: yes right, removed it now,
 i had thought to remove it in next refactor PR.

File: testsupport/wait/awaitility.go [hunk 5]
```diff
@@ -161,38 +156,13 @@ func (a *Awaitility) WaitForService(t *testing.T, name string) (corev1.Service,
 // WaitForToolchainClusterWithCondition waits until there is a ToolchainCluster representing a operator of the given type
 // and running in the given expected namespace. If the given condition is not nil, then it also checks
 // if the CR has the ClusterCondition
```
Comment (MatousJobanek) [inline testsupport/wait/awaitility.go:158]: could you please also update the comment?

**Review by MatousJobanek — APPROVED** [review]
Looks good :+1: 
Great job :1st_place_medal:

**Pull request merged**